### PR TITLE
PR: Delete buddy press pending users

### LINKED ISSUE #151: Delete Users by User Role - Delete Buddypress 'pending' users
'pending' users are described here - https://buddypress.org/support/topic/how-to-mass-delete-pending-users-1-million-users-i-already-did-phpmyadmin/#post-231732

Request came through support forum

### PR DESCRIPTION
**Pull request by esudalaimadan**
Delete buddy press pending users module

Fix #151

### CODE REVIEW

File: include/Core/BulkDelete.php [hunk 1]
```diff
@@ -474,6 +475,9 @@ private function get_delete_users_admin_page() {
 
 		$users_page->add_module( new DeleteUsersByUserRoleModule() );
 		$users_page->add_module( new DeleteUsersByUserMetaModule() );
+		if ( class_exists( 'BuddyPress' ) ) {
```
Comment (sudar) [inline include/Core/BulkDelete.php]: Add an additional check to make sure that this is a single site install and not multisite.

**Review by sudar — CHANGES_REQUESTED** [review]
Left a simple comment. Once it is fixed, this can be merged.

File: include/Core/BulkDelete.php (hunk 1 above)
Comment (esudalaimadan) [reply include/Core/BulkDelete.php]: Fixed.

**Comment by sudar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- include/Core/Users/Modules/DeleteBPPendingUsersModule.php  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271&bid=16151949)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			/* translators: 1 Number of users deleted */
+			'deleted_multiple' => __( 'Deleted %d Buddy Press pending users', 'bulk-delete' ),
+		);
+	}
+
+	/**
+	 * Render delete users box.
+	 */
+	public function render() {
+		$count = \BP_Signup::count_signups();
```
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php]: Codacy Issue found: [Avoid using static access to class '\BP_Signup' in method 'render'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+	// phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+	protected function append_to_js_array( $js_array ) {
+		$js_array['validators'][ $this->action ] = 'noValidation';
+
+		return $js_array;
+	}
+
+	// phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+	protected function build_query( $options ) {
```
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php]: Codacy Issue found: [Avoid unused parameters such as '$options'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+			</table>
+		</fieldset>
+		<!-- Users end-->
+
+		<?php
+		$this->render_submit_button();
+	}
+
+	// phpcs:ignore Squiz.Commenting.FunctionComment.Missing
+	protected function convert_user_input_to_options( $request, $options ) {
```
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php]: Codacy Issue found: [Avoid unused parameters such as '$request'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		foreach ( $ids as $id ) {
+			$deleted = wp_delete_user( $id );
+			if ( $deleted ) {
+				$count++;
+			}
+		}
+
+		$table_name = $wpdb->prefix . 'signups';
+		$date_query = str_replace( 'user_registered', 'registered', $date_query );
+		$status     = $wpdb->get_col( $wpdb->prepare( "DELETE FROM {$table_name} WHERE active = %d {$date_query}", 0 ) );
```
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php]: Codacy Issue found: [Avoid unused local variables such as '$status'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

Comment (sudar) [reply include/Core/Users/Modules/DeleteBPPendingUsersModule.php]: @esudalaimadan 

The assignment to ``$status`` is dead and can be removed.

**Review by sudar — CHANGES_REQUESTED** [review]
@esudalaimadan 

One more small change regarding the `$status` variable.

**Comment by sudar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- include/Core/Users/Modules/DeleteBPPendingUsersModule.php  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271&bid=16151949)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php (hunk 3 above)
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php:70]: Codacy Issue found: [Avoid unused parameters such as '$options'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php (hunk 4 above)
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php:58]: Codacy Issue found: [Avoid unused parameters such as '$request'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

File: include/Core/Users/Modules/DeleteBPPendingUsersModule.php (hunk 2 above)
Comment (sudar) [inline include/Core/Users/Modules/DeleteBPPendingUsersModule.php:38]: Codacy Issue found: [Avoid using static access to class '\BP_Signup' in method 'render'.](https://app.codacy.com/manual/BulkWP/bulk-delete/pullRequest?prid=4872271)

**Review by sudar — APPROVED** [review]

**Pull request merged**